controllers/install/eventingester_controller.go

```diff
@@ -69,27 +81,62 @@ func (r *EventIngesterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, err
 	}
 
+	deletionTimestamp := eventIngester.ObjectMeta.DeletionTimestamp
+	// examine DeletionTimestamp to determine if object is under deletion
+	if deletionTimestamp.IsZero() {
+		// The object is not being deleted, so if it does not have our finalizer,
+		// then lets add the finalizer and update the object. This is equivalent
+		// registering our finalizer.
+		if !controllerutil.ContainsFinalizer(&eventIngester, eventIngesterFinalizer) {
+			logger.Info("Attaching finalizer to EventIngester object", "finalizer", eventIngesterFinalizer)
+			controllerutil.AddFinalizer(&eventIngester, eventIngesterFinalizer)
+			if err := r.Update(ctx, &eventIngester); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		logger.Info("EventIngester object is being deleted", "finalizer", eventIngesterFinalizer)
+		// The object is being deleted
+		if controllerutil.ContainsFinalizer(&eventIngester, eventIngesterFinalizer) {
+			// our finalizer is present, so lets handle any external dependency
+			logger.Info("Running cleanup function for EventIngester object", "finalizer", eventIngesterFinalizer)
+
+			// remove our finalizer from the list and update it.
+			logger.Info("Removing finalizer from EventIngester object", "finalizer", eventIngesterFinalizer)
+			controllerutil.RemoveFinalizer(&eventIngester, eventIngesterFinalizer)
+			if err := r.Update(ctx, &eventIngester); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+
+		// Stop reconciliation as the item is being deleted
+		return ctrl.Result{}, nil
+	}
+
 	mutateFn := func() error { return nil }
 
+	logger.Info("Upserting EventIngester ServiceAccount object")
```

Comment:
I think we should probably move these logs into the if check.  Technically if ServiceAccount isn't specified, this won't be upserted.